src/openms/include/OpenMS/KERNEL/AreaIterator.h

```diff
@@ -275,6 +276,12 @@ namespace OpenMS
         return p_.current_scan_->getDriftTime();
       }
 
+      /// returns the current scan into which the iterator points
+      const SpectrumT& getSpectrum() const
```

Comment:
@PierreSnell I added a getSpectrum member to AreaIterator.

Now you can just access the ion mobility for an iterator with
``` 
auto peak_index = it.getPeakIndex().peak_index;
auto peak_IM = it.getSpectrum().getFloatDataArray[im_index][peak_index];
```

Reply:
note that im_index needs to be determined with getIMData

Reply:
Just updated to take that into account. Hope it make sense with what you expected.